scanner.go

```diff
@@ -241,13 +213,12 @@ func (s *scanner) request() (*pb.ScanResponse, hrpc.RegionInfo, error) {
 
 // update updates the scanner for the next scan request
 func (s *scanner) update(resp *pb.ScanResponse, region hrpc.RegionInfo) {
-	if resp.GetMoreResultsInRegion() {
-		if resp.ScannerId != nil {
-			s.scannerID = resp.GetScannerId()
-		}
-	} else {
+	if s.isRegionScannerClosed() && resp.ScannerId != nil {
+		s.openRegionScanner(resp.GetScannerId())
+	}
+	if !resp.GetMoreResultsInRegion() {
 		// we are done with this region, prepare scan for next region
-		s.scannerID = noScannerID
+		s.closeRegionScanner()
```

Comment:
so does this mean that for every scanned region, it's going to send an RPC to regionserver to close the scanner and regionserver doesn't automatically close those? I guess my understanding was wrong when I implemented this.

Reply:
The server would close them after timing out, which is 5 min later by default, wasting resources and throwing warnings. For a cluster with heavy load of small scans, it is not scalable.

Reply:
client should do best effort to call close as soon as done. Server closing scanner is the last resort.

Reply:
Wondering if regionserver could be a bit smarter here as it knows that the scanner for the region has been exhausted and close it automatically rather than waiting for timeout (but that's a server side change). Thanks for explanation.